### PR TITLE
#11789: Fix firmware/kernel padding/alignment

### PR DESCRIPTION
**Pull request by pgkeller**
### Ticket
# 11789
### Problem description
Padding was wrong for erisc

### What's changed
Fixed padding for erisc, simplified mechanism

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tt_metal/hw/toolchain/sections.ld [hunk 1]
```diff
@@ -146,7 +146,7 @@ SECTIONS
 #ifdef TYPE_FIRMWARE
   /* FW must align to 16 byte boundary so Kernel begins aligned to meet noc
```
Comment (nathan-TT) [inline tt_metal/hw/toolchain/sections.ld:147]: This comment is no longer true -- move it to where the appropriate #define is?

File: tt_metal/hw/inc/grayskull/dev_mem_map.h [hunk 2]
```diff
@@ -103,7 +103,7 @@
 /////////////
 // Padding/alignment restrictions needed in linker scripts
 // Note we pad then align so, for example, erisc can keep space between fw/kernels
```
Comment (nathan-TT) [inline tt_metal/hw/inc/grayskull/dev_mem_map.h:105]: and these comments too? padding of zero is not interesting

**Review by nathan-TT — COMMENTED** [review]
some comments are now stale, but the math(s) is right now

File: tt_metal/hw/inc/grayskull/dev_mem_map.h (hunk 2 above)
Comment (pgkeller) [reply tt_metal/hw/inc/grayskull/dev_mem_map.h:105]: yeah, good catch

File: tt_metal/hw/toolchain/sections.ld [hunk 3]
```diff
@@ -26,14 +26,15 @@
...
 #define MEM_PAD MEM_IERISC_KERNEL_PAD
 #else
-#define MEM_PAD MEM_TENSIX_KERNEL_PAD
+#define MEM_PAD 0
 #endif
+/* FW must align to 16 byte boundary so kernel begins aligned to meet noc alignment constraints */
+#define MEM_ALIGN 16
 #else
 #define MEM_PAD 0
+#define MEM_ALIGN 4
```
Comment (abhullar-tt) [inline tt_metal/hw/toolchain/sections.ld:35]: do we need this? from usage below it looks like `MEM_ALIGN` used is guarded under `#ifdef TYPE_FIRMWARE`?

Comment (pgkeller) [reply tt_metal/hw/toolchain/sections.ld:35]: one use is, other use is not

**Review by nathan-TT — APPROVED** [review]
lgtm

**Review by abhullar-tt — APPROVED** [review]

**Pull request merged**